tests/ui/transmute_undefined_repr.rs

```diff
@@ -109,6 +109,14 @@ fn main() {
         let _: Ty2<u32, u32> = transmute(value::<MaybeUninit<Ty2<u32, u32>>>()); // Ok
 
         let _: Ty<&[u32]> = transmute::<&[u32], _>(value::<&Vec<u32>>()); // Ok
+
+        let _: *const Ty2<u32, u32> = transmute(value::<*const Ty2C<Ty2<u32, u32>, u32>>()); // Ok
+        let _: *const Ty2C<Ty2<u32, u32>, u32> = transmute(value::<*const Ty2<u32, u32>>()); // Ok
```

Comment:
Why is this okay? This transmutes a `Ty2<u32, u32>` to a larger value. Wouldn't this mean that the last `u32` of `Ty2C` is undefined?

Reply:
This would be the same as down-casting a type. Not necessarily correct, but it's possible.

Reply:
I feel like that should be linted, but that's not necessarily the scope of this PR. :upside_down_face:

Reply:
It would be better as a different lint. This is for transmutes where none of the fields have a defined value because they could be in an unexpected order.